e2e/tests/stablex_test.rs

```diff
@@ -111,3 +120,124 @@ fn test_stablex() {
         .expect("Cannot get balance after");
     assert_eq!(balance_after - balance_before, 999_000.into())
 }
+
+ethcontract::contract!("dex-contracts/build/contracts/BatchExchange.json");
+
+#[test]
+fn test_rinkeby() {
+    // Setup instance and default tx params
+    let (eloop, http) = Http::new("https://node.rinkeby.gnosisdev.com/").expect("transport failed");
+    eloop.into_remote();
+    let web3 = Web3::new(http);
+    let mut instance = BatchExchange::deployed(&web3)
+        .wait()
+        .expect("Cannot get deployed Batch Exchange");
+    let secret = {
+        let private_key: H256 = env::var("PK")
+            .expect("PK env var not set")
+            .parse()
+            .expect("PK not parseable");
+        SecretKey::from_raw(&private_key[..])
+            .map_err(ethsign::Error::from)
+            .expect("Cannot derive key")
+    };
+    let account = Account::Offline(secret, None);
+    instance.defaults_mut().from = Some(account.clone());
+    instance.defaults_mut().gas = Some(1_000_000.into());
+    instance.defaults_mut().gas_price = Some(8_000_000_000u64.into());
+
+    let nonce = web3
+        .eth()
+        .transaction_count(account.address(), None)
+        .wait()
+        .expect("Cannot get nonce");
+    println!("Using account {:x} with nonce {}", account.address(), nonce);
+
+    // Gather token and batch info
+    let token_a = instance
+        .token_id_to_address_map(0)
+        .call()
+        .wait()
+        .expect("Cannot get first Token address");
+    let token_b = instance
+        .token_id_to_address_map(7)
+        .call()
+        .wait()
+        .expect("Cannot get second Token address");
+    let batch = instance
+        .get_current_batch_id()
+        .call()
+        .wait()
+        .expect("Cannot get batchId");
+
+    // Approve Funds
+    let first_approve = IERC20::at(&web3, token_a)
+        .approve(instance.address(), 1_000_000.into())
+        .nonce(nonce)
+        .gas(1_000_000.into())
+        .gas_price(8_000_000_000u64.into())
+        .from(account.clone())
+        .send_and_confirm(Duration::from_secs(1), 1);
+    let second_approve = IERC20::at(&web3, token_b)
+        .approve(instance.address(), 1_000_000.into())
+        .nonce(nonce + 1)
+        .gas(1_000_000.into())
+        .gas_price(8_000_000_000u64.into())
+        .from(account)
+        .send_and_confirm(Duration::from_secs(1), 1);
+
+    // Deposit Funds
+    let first_deposit = instance
+        .deposit(token_a, 1_000_000.into())
+        .nonce(nonce + 2)
```

Comment:
Is speficying `nonce` something that is necessary for the user to do manually with ethcontracts?

Reply:
We need it here because we want to send multiple txs without waiting for the previous one to be confirmed (save a lot of time). 
Thus we need to manually increment the nonce for each tx. By default ethcontract-rs checks "how many confirmed txs has this account sent in the past" and then sets nonce as this value + 1.

Reply:
VERY COOL!